overlays/cross-compilation.nix

```diff
@@ -5,6 +5,41 @@
 #
 {...}: {
   nixpkgs.overlays = [
-    (final: prev: {})
+    (final: prev: let
+      crossCompiling = prev.stdenv.buildPlatform != prev.stdenv.hostPlatform;
```

Comment:
I think we might not necessarily need to do this since we are anyway including this overlay only on cross-compiled builds

Reply:
Correct, but we don't need this patch applied for buildPackages (which happens when we apply patch unconditionally here), but only for target build